byway/src/main/java/com/google/sps/servlets/DestinationsServlet.java

```diff
@@ -62,7 +61,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException { 
-    Entity entity;
+    Entity entity = null;
     try{ 
       entity = datastore.get(userKey) ;
     } catch(EntityNotFoundException e){
```

Comment:
I think you can break down some code after the catch statement. For instance, when checking if the destinations array is null, you could do something like: `ArrayList<String> destinations = (ArrayList<String>) entity.getProperty("destinations")`. Then, you can check, ie `if(destinations == null)`, then you can make a new ArrayList like in line 78. Then after that, you can add to the destinations and set properties.

Reply:
done